x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java

```diff
@@ -148,9 +146,14 @@ public void doClose() {
 
     static class Metrics implements BucketedSort.ExtraData, Releasable {
         private final MetricValues[] values;
+        private final List<String> names;
 
         Metrics(MetricValues[] values) {
             this.values = values;
+            names = new ArrayList<>(values.length);
+            for (MetricValues value : values) {
```

Comment:
I agree that the for loop is more readable here than the stream.

Reply:
And the stream was creating a list with size 10  although we were adding only one element.